### PR TITLE
Miscellaneous minor fixups and refactors

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoIdentifierLeftChecker.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import static java.lang.String.format;
+
+public final class NoIdentifierLeftChecker
+        implements PlanSanityChecker.Checker
+{
+    @Override
+    public void validate(PlanNode plan, Session session, Metadata metadata, SqlParser sqlParser, Map<Symbol, Type> types)
+    {
+        for (Expression expression : ExpressionExtractor.extractExpressions(plan)) {
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoIdentifierLeftChecker.java]: use: `ExpressionTreeUtils#extractExpressions(java.lang.Iterable<? extends com.facebook.presto.sql.tree.Node>, java.lang.Class<T>)` and validate `isEmpty()`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java [hunk 2]
```diff
@@ -87,6 +88,9 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
             if (call.getFilter().isPresent()) {
                 Expression filter = call.getFilter().get();
                 Symbol symbol = context.getSymbolAllocator().newSymbol(filter, BOOLEAN);
+                if (mask.isPresent()) {
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java]: remove, dead code.

File: presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java [hunk 3]
```diff
@@ -43,6 +43,15 @@ public ExpressionTreeRewriter(ExpressionRewriter<C> rewriter)
         this.visitor = new RewritingVisitor();
     }
 
+    public List<Expression> rewrite(List<Expression> items, Context<C> context)
```
Comment (findepi) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java]: This should probably be `private`.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoIdentifierLeftChecker.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            new DefaultTraversalVisitor<Void, Void>()
+            {
+                @Override
+                protected Void visitIdentifier(Identifier node, Void context)
+                {
+                    throw new IllegalStateException(format("Unexpected identifier in logical plan: %s", node));
+                }
+            }.process(expression, null);
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoIdentifierLeftChecker.java]: You can
```
AstUtils.preOrder(expression)
 .filter(Identifier.class::isInstance)
 .findFirst(.....)
```

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java [hunk 5]
```diff
@@ -311,10 +311,8 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Set<Symbol
                     Aggregation aggregation = entry.getValue();
                     FunctionCall call = aggregation.getCall();
                     expectedInputs.addAll(SymbolsExtractor.extractUnique(call));
-                    if (aggregation.getMask().isPresent()) {
-                        expectedInputs.add(aggregation.getMask().get());
-                    }
-                    aggregations.put(symbol, new Aggregation(call, aggregation.getSignature(), aggregation.getMask()));
+                    aggregation.getMask().ifPresent(expectedInputs::add);
+                    aggregations.put(symbol, aggregation);
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java]: Also inline `call` variable

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java (hunk 2 above)
Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java]: If this is dead, the maybe `verify(!mask.isPresent())`, especially that we create new mask 2 lines  below.

**Pull request merged**